enhancements/ingress/configurable-dns-loadbalancerservice.md

```diff
@@ -0,0 +1,316 @@
+  from OCP 3.x to 4.x where DNS was being managed externally.
+
+### Non-Goals
+
+- Provide a day-0 solution for cluster installations involving external DNS
+  management by the customer.
+- DNS migration from managed to unmanaged.
+
+## Proposal
+
+Currently, there is partial support provided by the `installer` (in UPI installations)
+to disable the `cluster-ingress-operator` from managing the DNS cluster-wide as documented
+[here](https://github.com/openshift/installer/blob/master/docs/user/aws/install_upi.md#remove-dns-zones). 
+
+The proposed solution adds support for more fine-grained control over specific
+ingresscontrollers on how they manage the wildcard DNS records associated with them.
+
+### Workflow Description
+
+This feature is designed as Day-2 solution and is geared towards cluster
+admins. It mainly applies to scenarios where specific ingresscontrollers
+need to be configured to not manage DNS records associated with them.
+
+__Note__: This feature is only supported on new or non-default ingresscontrollers.
+The default ingresscontroller can be modified/updated at the discretion of
+the cluster admin. 
+
+#### Modifying/Updating an ingresscontroller
+
+The ingresscontroller will need to be deleted and recreated with the updated domain
+present in if required domain is _not_ the same as cluster domain configured at
+*ingress.config.openshift.io/cluster* `.spec.domain`. If not needed, the workflow
+defined below is sufficient,
+- The new domain that needs to be associated with the ingresscontroller
+    must be created prior to making any changes.
+- The ingresscontroller must be edited to set `.loadBalancer.dns` to `Unmanaged`.
+- This will trigger a reconcile of the controller, resulting in updating
+    the following conditions on the ingress operator
+  - `DNSManaged` condition will be set to false and reason updated to
+    UnmanagedDNS.
+- The DNSRecord will also be updated as part of the same reconcile, 
+  `.spec.managementPolicy` will be set to `Unmanaged` and the following conditions
+  will be updated,
+  - `DNSReady` condition will be set to true and reason updated to
+    UnmanagedDNS.
+   
+    
+__Note__: Similarly, to create a new ingresscontroller where DNS is managed
+          externally, the same workflow can be followed. No *DNSRecord*s will be
+          created in the flow.
+
+### API Extensions
+
+The ingresscontroller CRD is extended by adding an optional field `DNS`
+of type `LoadBalancerDNSManagementPolicy`. This field will default to `Managed`.
+
+```go
+// LoadBalancerStrategy holds parameters for a load balancer.
+type LoadBalancerStrategy struct {
+    // <snip>
+
+    // dns indicates if the lifecyle of the wildcard DNS record
+    // associated with the load balancer service will be managed by
+    // the ingress operator.
+    //
+    // +kubebuilder:default:="Managed"
+    // +kubebuilder:validation:Optional
+    // +kubebuilder:validation:Enum=Managed;Unmanaged
```

Comment:
I would clear this with the API team before changing but I did get a few bits of feedback on this sort of approach with my API PR:
* [Describe what an empty value means](https://github.com/openshift/api/pull/1110#discussion_r818889153)
* [Additional Discussion](https://github.com/openshift/api/pull/1110#pullrequestreview-925198254)

Reply:
Yeah for sure. To be clear I'm only sharing for the sake of sharing; not necessarily anything actionable here :smile: